disruption_py/machine/d3d/physics.py

```diff
@@ -407,15 +407,25 @@ def get_density_parameters(params: PhysicsMethodParams):
         ption-py/blob/matlab/DIII-D/get_density_parameters.m)
         - pull requests: #[249](https://github.com/MIT-PSFC/disruption-py/pull/249)
         """
-        ne, t_ne = params.mds_conn.get_data_with_dims(
-            r"\density", tree_name="_efit_tree"
-        )
+        try:
+            ne, t_ne = params.mds_conn.get_data_with_dims(
+                r"\density", tree_name="_efit_tree"
+            )
+        except mdsExceptions.MdsException:
+            ne = [np.nan]
+            t_ne = [np.nan]
         # If EFIT disruption tree does not contain density data,
         # then read density from BCI subtree of D3D main tree
-        # TODO: Find a shot to test this logic
-        if len(~np.isnan(ne)) == 0:
+        # Example: 170480:
+        #  - r"\density" gives ne = array([], shape=(4, 0), dtype=float64),
+        #  - r"\denv2" gives actual density data
+        if sum(~np.isnan(ne)) == 0:
```

Comment:
I haven't tested it, but what about something like this?
```suggestion
        if not np.any(np.isfinite(ne)):
```